File: Adm/Models/src/Local.java
package Adm.Models.src;

import java.util.HashMap;
import java.util.Map;

public class Local implements Inter {
  private int id;
  private String cidade;

  public Local(int id, String cidade) {
    setId(id);
    setCidade(cidade);
  }

  @Override
  public String toString() {
    return "Local:"+
      "id=" + this.id +
      ", cidade=" + this.cidade;
  }

  public Map<String, Object> toDict() {
    Map<String, Object> dict = new HashMap<>();
    dict.put("id", this.id);
    dict.put("cidade", this.cidade);
    return dict;
}

  public void setId(int id) {
    if (id >= 0) {
      this.id = id;
    } else {
      throw new IllegalArgumentException("ID inválido");
    }
  }

  public void setCidade(String cidade) {
    if (cidade instanceof String) {
      this.cidade = cidade;
    } else {
      throw new IllegalArgumentException("Cidade inválida");
    }
  }

  public int getId(){
    return this.id;
  }

  public String getCidade() {
    return this.cidade;
  }
}

File: Adm/Models/Local.cs
using System;
using System.Collections.Generic;

namespace Adm.Models
{
    public class Local : Inter
    {
        // Atributos privados
        private int id;
        private string cidade;

        public Local(int id, string cidade)
        {
            setId(id);
            setCidade(cidade);
        }

        // Métodos setters com validação
        public void setId(int id)
        {
            if (id < 0)
            {
                throw new ArgumentException("ID inválido");
            }
            this.id = id;
        }

        public void setCidade(string cidade)
        {
            if (string.IsNullOrWhiteSpace(cidade))
            {
                throw new ArgumentException("cidade não pode ser vazio");
            }
            this.cidade = cidade;
        }


     
        // Métodos getters
        public int getId() => this.id;
        public string getCidade() => this.cidade;

        
        public Dictionary<string, object> ToDict()
        {
            return new Dictionary<string, object>
            {
                { "id", this.id },
                { "cidade", this.cidade}
               };
            }

        // Método ToString
        public override string ToString()
        {
            return $"Cliente: id={this.id}, cidade={this.cidade}";
        }
    }
}
        
